legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/logicalplan/datasets/StagedFilesDatasetProperties.java

```diff
@@ -35,11 +35,11 @@ default void validate()
     {
         if (filePatterns().size() > 0 && filePaths().size() > 0)
         {
-            throw new IllegalArgumentException("Cannot build StagedFilesDatasetProperties, Only one out of filePatterns and filePaths should be provided");
+            throw new IllegalStateException("Cannot build StagedFilesDatasetProperties, Only one out of filePatterns and filePaths should be provided");
         }
         if (filePatterns().size() == 0 && filePaths().size() == 0)
         {
-            throw new IllegalArgumentException("Cannot build StagedFilesDatasetProperties, Either one of filePatterns and filePaths must be provided");
+            throw new IllegalStateException("Cannot build StagedFilesDatasetProperties, Either one of filePatterns and filePaths must be provided");
```

Comment:
Should these be illegal state or illegal arguments? When someone configures bulk load, file patterns and paths are something users can pass in